### DISCUSSION TITLE
Plan for MoE

### DISCUSSION THREAD

**Discussion by nilsjohanbjorck** [NVIDIA-NeMo/NeMo#5472] [Ideas]
Is there a plan for adding Mixture of experts for GPT-style models? I've found this PR https://github.com/NVIDIA/NeMo/pull/5409/files but that seems to be for T5-like models. Thanks!

**Comment by okuchaiev** [comment]
@aklife97 can you please share what is coming for MoE?

**Comment by abhinav-khattar** [comment]
Adding MoE for GPT is pretty straightforward. I'll create a PR for MoE for GPT soon after some testing. I'll update this thread once its added

**Comment by adamlin120** [comment]
@aklife97  I am wondering is there any plan to include expert parallel to the MoE? Any guidance to support expert parallelism would be super helpful. Thanks

**Comment by abhinav-khattar** [comment]
Hey @adamlin120 expert-parallel for MoE is high priority, we have it working with Megatron-core and plan to support it inside NeMo in a few months

**Comment by DevHector-Bit** [comment]
Its possible save checkpoint from Moe models? i got some issues trying to save one the only way i se now its saving weights but doesnt work for resume training
